### PR TITLE
conditional keystone user cherry pick for 3.0.x

### PR DESCRIPTION
**Pull request by angstwad**

### CODE REVIEW

File: roles/keystone-setup/tasks/main.yml [hunk 1]
```diff
@@ -55,7 +55,8 @@
       password: "{{ secrets.admin_password }}"
   with_items: "{{ keystone.users }}"
   # If cloud_admin already exists, don't set the cloud_admin user/pass (overwriting existing userpass)
```
Comment (sandfish8) [inline roles/keystone-setup/tasks/main.yml]: comment needs update too

File: roles/keystone-setup/tasks/main.yml [hunk 2]
```diff
@@ -55,7 +55,8 @@
...
-  when: "item.name != 'cloud_admin' or (item.name == 'cloud_admin' and 'cloud_admin' not in user_facts.ansible_facts.openstack_users|map(attribute='username')|list)"
+  when: "item.name not in ['cloud_admin', 'monitor'] or
+         (item.name in ['cloud_admin', 'monitor'] and item.name not in user_facts.ansible_facts.openstack_users|map(attribute='username')|list)"
```
Comment (sandfish8) [inline roles/keystone-setup/tasks/main.yml]: I think the `(item.name in ['cloud_admin', 'monitor']` after the `or` is unneccessary

Comment (j2sol) [inline roles/keystone-setup/tasks/main.yml]: Looking at this again.

What we want is to create the user if it doesn't exist, but not modify the user if it does exist.

The first part of the conditional, that may actually prevent the initial creation, if I'm reading that right.

Paul, can you unroll this logical set in plain english to walk us through it?

Comment (angstwad) [inline roles/keystone-setup/tasks/main.yml]: Plain english translation:

If the username is not cloud_admin or monitor, create/update user.  If the name is cloud_admin or monitor, and the user does not exist, create user.

Is that what you now see with the plain english description?  I can put that in there as a comment too if that helps.
